processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -447,13 +454,21 @@ public RowSignature resultArraySignature(TimeseriesQuery query)
     return result -> {
       final TimeseriesResultValue holder = result.getValue();
       final Map<String, Object> values = new HashMap<>(holder.getBaseObject());
-      if (calculatePostAggs && !query.getPostAggregatorSpecs().isEmpty()) {
-        // put non finalized aggregators for calculating dependent post Aggregators
-        for (AggregatorFactory agg : query.getAggregatorSpecs()) {
-          values.put(agg.getName(), holder.getMetric(agg.getName()));
+      if (calculatePostAggs) {
+        if (!query.getPostAggregatorSpecs().isEmpty()) {
+          // put non finalized aggregators for calculating dependent post Aggregators
+          for (AggregatorFactory agg : query.getAggregatorSpecs()) {
+            values.put(agg.getName(), holder.getMetric(agg.getName()));
+          }
+          for (PostAggregator postAgg : query.getPostAggregatorSpecs()) {
+            values.put(postAgg.getName(), postAgg.compute(values));
+          }
         }
-        for (PostAggregator postAgg : query.getPostAggregatorSpecs()) {
-          values.put(postAgg.getName(), postAgg.compute(values));
+        // Timeseries query has timestamp_floor expression on the timestamp dimension so we need to
+        // map the results to another dimension using the name (String) supplied by context key
```

Comment:
This comment doesn't make a lot of sense to me. How about something like this instead:

> If "timestampResultField" is set, we must include a copy of the timestamp in the result. This is used by the SQL layer when it generates a Timeseries query for a group-by-time-floor SQL query. The SQL layer expects the result of the time-floor to have a specific name that is not going to be "__time".

Reply:
Sounds good. Done